app/models/manageiq/providers/autosde/inventory/collector/storage_manager.rb

```diff
@@ -26,6 +25,44 @@ def storage_resources
     end
   end
 
+  def san_addresses
+    @san_addresses ||= @manager.autosde_client.StorageHostApi.storage_hosts_get.map do |addresses|
```

Comment:
Since you're making two calls to `@manager.autosde_client.StorageHostApi.storage_hosts_get` maybe we should cache that?

Also a little confusing to call this `addresses` when below you're calling it `host_initiators`

Recommend:
```
def storage_hosts
  @storage_hosts ||= @manager.autosde_client.StorageHostApi.storage_hosts_get
end

def san_addresses
  @san_addresses ||= storage_hosts.map do |storage_host|
  end
end

def host_initiators
  @host_initiators ||= storage_hosts.map ...
  end
end
```

Honestly these look less like "collector" methods and more like "parser" methods.  Can you move `san_addresses` and `host_initiators` to the Parser?  That would leave us with just `def storage_hosts` in the collector

Reply:
done